grafana_collector/grafana/dashboard.go

```diff
@@ -67,22 +95,212 @@ type dashContainer struct {
 	}
 }
 
+// MetircResult represents templating variable metric result
+type MetircResult struct {
+	Status string
+	Data   []map[string]interface{}
+}
+
+func unique(input []string) []string {
+	result := make([]string, 0, len(input))
+	m := make(map[string]bool)
+
+	for _, item := range input {
+		if _, ok := m[item]; !ok {
+			m[item] = true
+			result = append(result, item)
+		}
+	}
+
+	return result
+}
+
+func (d *Dashboard) getTemplatingVariableValue(tv TemplatingVariable) ([]string, error) {
+	re := regexp.MustCompile(`label_values\((\w+),\s*(\w+)\)$`)
```

Comment:
Make it a global variable, only compile it once, and I suggest add unit test for it